crates/source_analyzer/src/struct_hierarchy.rs

```diff
@@ -0,0 +1,113 @@
+use std::collections::BTreeMap;
+
+use syn::Type;
+use thiserror::Error;
+
+#[derive(Debug, PartialEq)]
+pub enum StructHierarchy {
+    Struct {
+        fields: BTreeMap<String, StructHierarchy>,
+    },
+    Optional {
+        child: Box<StructHierarchy>,
+    },
+    Field {
+        data_type: Type,
+    },
+}
+
+impl StructHierarchy {
+    pub fn new_struct() -> Self {
+        Self::Struct {
+            fields: Default::default(),
+        }
+    }
+}
```

Comment:
Implement `Default` instead?

Reply:
Done, also changed to derived default at all the call sites.